pyhive/hive.py

```diff
@@ -252,6 +258,41 @@ def _fetch_more(self):
         for row in response.results.rows:
             self._data.append([_unwrap_col_val(val) for val in row.colVals])
 
+    def poll(self, fetch_logs=False):
+        req = ttypes.TGetOperationStatusReq(
+            operationHandle=self._operationHandle
+        )
+        response = self._connection.client.GetOperationStatus(req)
+        _check_status(response)
+        
+        
+        
+        if fetch_logs:
+            req = ttypes.TFetchResultsReq(
+                operationHandle=self._operationHandle,
+                orientation=0,
+                maxRows=10000, 
+                fetchType=1
+            )
+            log_response = self._connection.client.FetchResults(req)
+            _check_status(log_response)
+            
+            del self._messages[:]
+            for row in log_response.results.rows:
+                self._messages.append([_unwrap_col_val(val) for val in row.colVals])
+        
+        return response.operationState
+    
+    def cancel(self):
```

Comment:
As per PEP-249 (DB API 2.0), cursor.close():
>    Close the cursor now (rather than whenever __del__ is called).
>
>    The cursor will be unusable from this point forward; an Error (or subclass) exception will be raised if any operation is attempted with the cursor.

In the current implementation, cursor.close() causes a state reset and further operations are allowed. This goes against specification.

In the Hive Thrift protocol, there is no notion of cursors, but only a session. Also, an operation can be cancelled and the session can be used again.

I think I prefer a separate cancel method. That way, at some point we can fix (or not) the behaviour of close to stick to DB-API specification.

Having said that, I am open to further discussion :)